src/config.rs

```diff
@@ -131,11 +132,17 @@ struct AppConfigInner {
     secure: bool,
     host: String,
     addr: SocketAddr,
+    pool: HttpRequestPool,
 }
 
 impl AppConfig {
     pub(crate) fn new(secure: bool, addr: SocketAddr, host: String) -> Self {
-        AppConfig(Rc::new(AppConfigInner { secure, addr, host }))
+        AppConfig(Rc::new(AppConfigInner {
+            secure,
+            addr,
+            host,
+            pool: HttpRequestPool::default(),
```

Comment:
AppConfig is per connection. Pool belongs in AppInitService

Reply:
AppConfig is passed as HttpService's factory config. It's constructed only once on every worker thread.(Unless worker is faulted and get reconstruct on a newly spawned thread.)

Reply:
That said it would get constructed based on service which means bind to different port would result in duplicate app config on each thread. But this is true for the old pool too.

Reply:
you're right, i've traced it through and it is indeed one per worker